server/src/main/java/io/deephaven/server/table/ops/filter/FilterFactory.java

```diff
@@ -139,7 +139,7 @@ private WhereFilter generateNumericConditionFilter(CompareCondition.CompareOpera
                 valueString = Long.toString(value.getLongValue());
                 break;
             case NANO_TIME_VALUE:
-                valueString = Long.toString(value.getNanoTimeValue());
+                valueString = String.format("'%d'", value.getNanoTimeValue());
```

Comment:
I'm worried about the multiple code paths for changing Literal into String. Can we delegate to `FilterPrinter.printNoEscape`? I'm noticing different logic for `DOUBLE_VALUE` as well...

Reply:
"multiple code paths for changing Literal into String" is unfortunately a required reality, even if you only count print and printNoEscape - there's no sane language that exists that covers "expressions you can pass in dh column language" and "expressions you can give to a match filter".

That said, printNoEscape is supposed to be the "expressions you can give to a match filter", and since this bug originated in range filters changing to be more like match filters (i am not at all convinced this is a good thing, match filters are terrible, but consistency is worth something i guess), I think you're right - this should go through FilterPrinter.

That said.... printNoEscape shouldn't matter for nanosecond timestamps, right? As long as it is a Literal, and since the new tweak in FilterPrinter handles such numeric literals in single quotes... it shouldn't technically matter if it is escaped or not. The signature of print(Condition) vs printNoEscape(Literal) does matter here though.

So yes, I agree, we should delegate in both places now to printNoEscape.